File: ga/4omini_result_2024_7/result/macro_func_mini/13_macro_func.cpp
use std::io;

macro_rules! cppmax {
    ($x:expr, $y:expr) => {
        if $x + $y < 5 { $y } else { $x }
    };
}

fn main() {
    let mut input = String::new();

    println!("Enter two integers:");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let nums: Vec<i32> = input
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("Please enter a valid integer"))
        .collect();

    if nums.len() != 2 {
        eprintln!("Please enter exactly two integers.");
        return;
    }

    let a = nums[0];
    let b = nums[1];

    let result = cppmax!(a, b);
    println!("Result: {}", result);
}

File: ga/gpt3_5_result_2024_6/result/macro_func_gpt/13_macro_func.rs
use std::io;

fn cppmax(x: i32, y: i32) -> i32 {
    if x + y < 5 {
        return y;
    } else {
        return x;
    }
}

fn main() {
    let mut input = String::new();

    println!("Enter two numbers:");
    io::stdin().read_line(&mut input).unwrap();
    let numbers: Vec<i32> = input.trim().split_whitespace().map(|x| x.parse().unwrap()).collect();

    let a = numbers[0];
    let b = numbers[1];

    let result = cppmax(a, b);
    println!("Result: {}", result);
}